### ISSUE TITLE
How to reproduce Table 1

### ISSUE THREAD

**Issue by yuzhenmao** [esteng/pragmatic_calibration#1]
Thank you for the amazing work! I’m trying to reproduce Table 1 from the original paper, but noticed that the results are not consistent with those reported in the paper. Could you let me know the exact command and hyper-parameter you used for those experiments? Thanks!

For example, for the Llama3-8B / chat+LACIE model, the results in the paper are:

AUROC: 0.72± 0.02
ECE: 0.12± 0.02
Precision: 0.70± 0.02
Recall: 0.83± 0.04
Abstained: 8.47± 1.03

The script I used for reproducing:

1. `CUDA_VISIBLE_DEVICES=0,1 bash scripts/dpo/trivia_qa/main/train_llama_3_on_10k_balanced.sh 369`

.../train_llama_3_on_10k_balanced.sh:

```
#!/bin/bash
seed=$1
llama_ckpt="meta-llama/Meta-Llama-3-8B-Instruct"

python trained_calibration/rl/train/train_dpo.py \
	--output_dir models/train_llama_8b_10k_balanced_long_${seed}_seed_chat \
	--model ${llama_ckpt} \
	--reward_model mistralai/Mistral-7B-v0.1 \
	--eval_steps 40 \
	--warmup_steps 10 \
	--save_steps 40 \
	--train_dataset data/trivia_qa/tqa_10k_train.jsonl \
	--valid_dataset data/trivia_qa/tqa_full_valid.jsonl \
	--valid_limit 500 \
	--per_device_train_batch_size 6 \
	--gradient_accumulation_steps 10 \
	--per_device_eval_batch_size 2 \
	--n_eval_batches 30 \
	--max_length 200  \
	--max_steps 250 \
	--seed ${seed} \
	--balance_types true
```

2. `CUDA_VISIBLE_DEVICES=0,1 bash scripts/dpo/trivia_qa/eval_llama_3_on_valid.sh ~/pragmatic_calibration/models/train_llama_8b_10k_balanced_long_369_seed_chat/checkpoint-250`

eval_llama_3_on_valid.sh:
```

llama_ckpt="mistralai/Mistral-7B-v0.1"

python trained_calibration/rl/evaluate_dpo.py \
    --model ${llama_ckpt} \
    --trained_model $1 \
    --data_path "data/trivia_qa/tqa_validation_mistral_v1_small.jsonl" \
    --limit -1 \
    --out_path $(dirname ${1})/eval_dpo_on_valid.jsonl \
    --n_per_prompt 1 \
    --seed 12 \
    --threshold 0.66
```

3. `python trained_calibration/eval/dpo_eval.py --eval_file ~/pragmatic_calibration/models/train_llama_8b_10k_balanced_long_369_seed_chat/eval_dpo_on_valid.jsonl --threshold 0.66`

However, what I got after running three different seeds:

| Metric      | NoSkip  | Skip  |
|------------|---------------------|-------------------|
| Precision  | 0.54 ± 0.17         | 0.74 ± 0.06      |
| Recall     | 0.49 ± 0.04         | 0.49 ± 0.04      |
| F1 Score   | 0.51 ± 0.08         | 0.59 ± 0.03      |
| Accuracy   | 0.41 ± 0.09         | 0.62 ± 0.02      |
| AUROC      | 0.65 ± 0.08         | 0.65 ± 0.07      |
| ECE        | 0.19 ± 0.12         | 0.07 ± 0.01      |
| Abstention | 0.34 ± 0.14         | 0.34 ± 0.14      |

Details:

seed 369: 
noskip: "trained": {"precision": 0.7259887005649718, "recall": 0.5099206349206349, "f1": 0.599067599067599, "accuracy": 0.504, "auroc": 0.735487071172555, "ece": 0.07541604482548109, "abstenion": 0.208}
skip: "trained": {"precision": 0.7870370370370371, "recall": 0.51, "f1": 0.6189320388349515, "accuracy": 0.6369426751592356, "auroc": 0.6944982456140351, "ece": 0.06446524123856975, "abstenion": 0.208}

seed 693 
noskip: "trained": {"precision": 0.40232558139534885, "recall": 0.5164179104477612, "f1": 0.45228758169934635, "accuracy": 0.335, "auroc": 0.5840287285377623, "ece": 0.29777759920143415, "abstenion": 0.465}
skip: "trained": {"precision": 0.6732283464566929, "recall": 0.5150602409638554, "f1": 0.5836177474402731, "accuracy": 0.6252354048964218, "auroc": 0.5701398559060362, "ece": 0.08142938566372517, "abstenion": 0.465}

seed 936:
noskip: "trained": {"precision": 0.4858757062146893, "recall": 0.44675324675324674, "f1": 0.4654939106901218, "accuracy": 0.385, "auroc": 0.635755464048147, "ece": 0.21074743558963144, "abstenion": 0.36}
skip: "trained": {"precision": 0.75, "recall": 0.4476439790575916, "f1": 0.560655737704918, "accuracy": 0.60062893081761, "auroc": 0.6883167745393083, "ece": 0.06218398057814666, "abstenion": 0.36}

Thank you in advance!
